client/components/board/boardFactory.js

```diff
@@ -39,16 +41,25 @@ angular.module('OpenEXP')
 
         // connect to the board and start streaming if no errors
         var connect = (portName) => {
+          return new Promise((res, rej) => {
             ourBoard.connect(portName)
-                .then(boardSerial => {
+                .then(() => {
                     ourBoard.on('ready', () => {
                         ourBoard.streamStart();
                         console.log('Connected!')
+                        counter = 0;
+                        ourBoard.on('sample', (sample) => {
+                          // wstream.write(sample.auxData[0] + ',' + sample.auxData[2] + '\n');
+                          // console.log(sample.channelData[0] + '\n')
+                          storage[counter] = sample;
+                          counter+=1;
```

Comment:
no its just to give each sample a unique value.  ultimately, what we'd like to do is turn the storage object into a buffer so we don't overload the app with data as its running.  the buffer could be written to a file as the data gets pushed out of the buffer